### PR TITLE
Fix compiling for iOS

### PR DESCRIPTION
**Pull request by ryanmcgrath**
Block off the `animator` proxy pieces for `appkit`/`macos` to fix iOS compiling. Animation support was exploratory to begin with, and the iOS APIs are different enough that they'd need some thinking through anyway.

### CODE REVIEW

**Comment by ryanmcgrath** [comment]
@madsmtm Curious if you have any idea offhand why `cargo fmt` would be giving inconsistent results - i.e, on my local machine it has no changes, yet CI here is complaining.

(The sheer number of files changed is also, admittedly, concerning, lol)

**Comment by madsmtm** [comment]
Nightly vs. stable `rustfmt` ;)

**Comment by simlay** [comment]
> Nightly vs. stable `rustfmt` ;)

Huh. I am surprised by how much difference there is. Also, I like the `rustfmt` from stable more but I think that's because of the trailing commas on the last items in a struct/list/etc.

File: examples/ios-beta/readme.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Since this needs to run in an iOS simulator or on a device, you can't run it lik
 - Start a simulator (Simulator.app).
 - `cargo install cargo-bundle`
 - `cargo bundle --example ios-beta --no-default-features --features uikit,autolayout --target x86_64-apple-ios`
```
Comment (simlay) [inline examples/ios-beta/readme.md:9]: Shall we add this to CI in a new github job?

**Comment by madsmtm** [comment]
> Huh. I am surprised by how much difference there is. Also, I like the `rustfmt` from stable more but I think that's because of the trailing commas on the last items in a struct/list/etc.

Did this in https://github.com/ryanmcgrath/cacao/pull/29#issuecomment-1179749602 because it matched the previous output better (~= smaller diff). See `rustfmt.toml` for the nightly features.

File: examples/ios-beta/readme.md (hunk 1 above)
Comment (ryanmcgrath) [reply examples/ios-beta/readme.md:9]: Probably makes sense to do so, yeah - that way we'll know if I break it again ;)

**Comment by ryanmcgrath** [comment]
Hmmm, amusingly enough I've never considered how linking something iOS-specific on a GitHub action would work. If anyone's got ideas or done it before I'm all ears, otherwise I'll have to take a look this weekend - made two basic attempts but I'm probably missing something hella obvious.

**Comment by simlay** [comment]
> Hmmm, amusingly enough I've never considered how linking something iOS-specific on a GitHub action would work. If anyone's got ideas or done it before I'm all ears, otherwise I'll have to take a look this weekend - made two basic attempts but I'm probably missing something hella obvious.

```
= note: ld: framework not found UIKit
[123](https://github.com/ryanmcgrath/cacao/runs/7912338422?check_suite_focus=true#step:5:124)
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

This is a new CI error for me. I've not used the `action-rs/cargo` action for iOS targets but I have done stuff with [`cargo-bundle` and some annoying environment variables in a `Makefile`](https://github.com/simlay/uikit-sys/blob/8d31ad8946f8ec42ea5c6d226eb3ce0e53072c8e/.github/workflows/rust.yml#L57-L59).

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -58,9 +58,15 @@ jobs:
...
+    # that more robust tests/checking infrastructure should exist for this side
+    # of things as the iOS portion gets iterated on.
+    #
+    # (e.g, this at the moment will not catch invalid selector calls, like if an appkit-specific
+    # selector is used for something on iOS)
+    - uses: actions-rs/cargo@v1
+      with:
+        command: build
+        target: x86_64-apple-ios
+        args: --example ios-beta --no-default-features --features uikit,autolayout
```
Comment (simlay) [inline .github/workflows/ci.yml:72]: From what I can tell, the build error you got in CI is the same as what I got when I ran `cargo build --example ios-beta --no-default-features --features uikit,autolayout` (notice the lack of `--target x86_64-apple-ios`). I think it needs to be done in the `args` section similar to how the [`use-cross section of the `actions-rs/cargo` references](https://github.com/actions-rs/cargo#cross-compilation).

**Comment by ryanmcgrath** [comment]
Closing since this was addressed/rolled in #48.

**Pull request closed without merging**